packages/components/src/navigator/navigator-screen/component.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import { motion, MotionProps } from 'framer-motion';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, useState } from '@wordpress/element';
+import { useReducedMotion, useFocusOnMount } from '@wordpress/compose';
+import { isRTL } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	WordPressComponentProps,
+} from '../../ui/context';
+import { View } from '../../view';
+import { NavigatorContext } from '../context';
+import type { NavigatorScreenProps } from '../types';
+
+const animationEnterDelay = 0;
+const animationEnterDuration = 0.14;
+const animationExitDuration = 0.14;
+const animationExitDelay = 0;
+
+// Props specific to `framer-motion` can't be currently passed to `NavigatorScreen`,
+// as some of them would overlap with HTML props (e.g. `onAnimationStart`, ...)
+type Props = Omit<
+	WordPressComponentProps< NavigatorScreenProps, 'div', false >,
+	keyof MotionProps
+>;
+
+function NavigatorScreen( props: Props, forwardedRef: Ref< any > ) {
+	const { children, path, ...otherProps } = useContextSystem(
+		props,
+		'NavigatorScreen'
+	);
+
+	const prefersReducedMotion = useReducedMotion();
+	const [ currentPath ] = useContext( NavigatorContext );
+	const isMatch = currentPath.path === path;
+	const ref = useFocusOnMount();
+
+	// This flag is used to only apply the focus on mount when the actual path changes.
+	// It avoids the focus to happen on the first render.
+	const [ hasPathChanged, setHasPathChanged ] = useState( false );
+	useEffect( () => {
+		setHasPathChanged( true );
+	}, [ path ] );
+
+	if ( ! isMatch ) {
+		return null;
+	}
+
+	if ( prefersReducedMotion ) {
+		return (
+			<View ref={ forwardedRef } { ...otherProps }>
```

Comment:
What does `View` bring compared to `div`?

Reply:
`View` is the basic building block of the components library. Every component that has been build recently uses `View` under the hood

Reply:
Looking at the code of the `View` component, it's not clear to me what that does bring to high-level components?

Reply:
There are a few practical advantages. The main is that `View` is an Emotion "styled" component, and thus it inherently enables styling (through Emotion and its APIs) and polymorphism when used under the hood.

In the past, `View` was also used to allow styles to be cached/applied across `iframes` (now we changed approach) — but this still shows how having `View` as a core building block can be very important.

In my opinion, it is also conceptually important that we use `View` and the base components of all our components — every component is a `View` at its core.

Reply:
The main advantage to `View` in my "view" of things (lol sorry) is that it allows polymorphism for components. Other advantages are trivial. Styling any element is already possible through the `className` API and we don't use Emotion's CSS prop, so that's not really a reason to use the `View` component. I think it comes down to that it allows for polymorphism.